### ISSUE TITLE
fix everware re-start

### ISSUE THREAD

**Issue by anaderi** [everware/everware#109]
1. `make stop`
2. `make run-daemon`
3. login again, and supply URL https://github.com/everware/everware-dimuon-example
4. I get : `Error 500: Permission failure checking authorization, I may need to be restarted`
5. after ControlPanel -> Stop -> Spawn everything is fine

**Comment by anaderi** [comment]
seems that problem is not about restart, but about dockerspawner. After I get 'Error 500' once again, I've looked at containers running, and found old container named 'jupyter-anaderi' started many hours ago. So it seems that after image has been created, as we start container of given name, dockerspawner finds it running and skips creation.

**Comment by anaderi** [comment]
see #116

**Comment by betatim** [comment]
Shutting down `everware-server` should remove all containers. For me the log output prints something like "shutting down user servers". This is shutdown by pressing C-c in the terminal. Do you know what happens if you send SIGINT vs SIGKILL?

**Comment by astiunov** [comment]
Jupyterhub handles SIGINT and SIGTERM signals. It calls this routine: https://github.com/jupyter/jupyterhub/blob/b36cd92ae6940060fbaec69d23d56a1c13df9248/jupyterhub/app.py#L1103, which by default calls `stop()` for all running containers. If you send SIGKILL, `everware-server` would be killed immediately, so all containers are still running

**Comment by anaderi** [comment]
SIGINT is handled the same way as SIGKILL (i.e. everware just stops without stopping containers).

**Comment by astiunov** [comment]
No, `atexit` by default handles SIGINT, so SIGINT is handled with stopping all containers (that's why you can use C-c)
